backend/fms_core/services/report.py

```diff
@@ -20,9 +22,42 @@ class TimeWindow(TextChoices):
 
 REPORT_HEADER_COLOR = "34a8eb"
 
-def get_date_range_with_window(start_date: str, end_date: str, time_window: TimeWindow):
+class EnhancedName(TypedDict):
+    name: str
+    display_name: str
+
+class ReportInfo(TypedDict):
+    name: str
+    groups: list[EnhancedName]
+    time_windows: list[str]
+
+class ReportHeader(TypedDict):
+    name: str
+    display_name: str
+    field_order: int
+    aggregation: str
+    data_type: str
+
+class TimeWindowData(TypedDict):
+    time_window: datetime.date
+    time_window_label: str
+    time_window_start: datetime.date
+    time_window_end: datetime.date
+    time_window_data: list[dict[str, Union[str, int]]]
+    
+class ReportData(TypedDict):
+    name: str
```

Comment:
This type is a lie ;(
You don't provide `name` field. You provide `report`.
I wonder if this can be reinforced by using dataclass instead of a TypedDict ;)

Reply:
`dataclass` might be a bad idea actually...

Reply:
yeah indeed i fixed it ... i kind of put the wrong attribute on this type since it's essentially just fluff i missed it ;).